VanHalenSort/color_sense_Skittles_RGB_MQTTclientWIFI_topic.ino

```diff
@@ -0,0 +1,224 @@
+#include <WiFi101.h>
+#include <MQTT.h>
+#include <math.h>
+
+//TCS230 pins wiring to Arduino
+#define S0 4
+#define S1 5
+#define S2 6
+#define S3 7
+#define sensorOut 8
+
+//Setting up MQTT/WIFI connection
+const char WIFI_SSID[] = "XXX";
+const char WIFI_PASS[] = "XXX";
+const char mqttServer[] = "192.168.2.3";
+const int mqttServerPort = 1883;
+const char key[] = "key";
+const char secret[] = "secret";
+const char device[] = "mkr1000";
+int status = WL_IDLE_STATUS;
+WiFiClient net;
+MQTTClient client;
+unsigned long lastMillis = 0;
+
+//define colors
+enum Colors{NONE, RED, GREEN, PURPLE, YELLOW, ORANGE};
+////////////////////////////////////////////////////////////////////////////////////
+void setup() {
+  Serial.begin(9600);
+  connect();
+  
+  // Setting the outputs
+  pinMode(S0, OUTPUT);
+  pinMode(S1, OUTPUT);
+  pinMode(S2, OUTPUT);
+  pinMode(S3, OUTPUT);
+  
+  // Setting the sensorOut as an input
+  pinMode(sensorOut, INPUT);
+  
+  //Frequency scaling to 2%
+  digitalWrite(S0,LOW);
+  digitalWrite(S1,HIGH);
+  
+};
+////////////////////////////////////////////////////////////////////////////////
+void loop() {
+  client.loop();
+
+  if (!net.connected()) {
+    connect();
+  }
+  //MAIN PROGRAM
+  if (millis() - lastMillis > 1000) {
+    lastMillis = millis();
+    Serial.println("R: "+String(redSmoothOutFrequency())+"  G: "+String(greenSmoothOutFrequency())+"  B: "+String(blueSmoothOutFrequency()));
+    //THIS IS WHERE THE COLOR IS ACTUALLY DETECTED - whatColor() method; EACH CHANNEL IS SMOOTHED OUT (AV. OF 5 CONSECUTIVE MEASUREMENTS) FOR MORE STABLE RESULTS
+	client.publish("/color", whatColor(redSmoothOutFrequency(), greenSmoothOutFrequency(), blueSmoothOutFrequency()));
+    lastMillis = millis();
+  }
+
+};
+
+///////////////////////////////////////////////////////////////////////////////////
+const char* whatColor(int redFrequency, int greenFrequency, int blueFrequency){
+  Colors tempColor = NONE;
+
+  if((isInRange(56, 60, redFrequency)) && (isInRange(60, 63, greenFrequency)) && (isInRange(17, 20, blueFrequency))){
+    tempColor = YELLOW;
+  }
+  
+  if((isInRange(67, 71, redFrequency)) && (isInRange(83, 86, greenFrequency)) && (isInRange(21, 25, blueFrequency))){
+    tempColor = RED;
+  }
+ 
+  if((isInRange(84, 88, redFrequency)) && (isInRange(84, 88, greenFrequency)) && (isInRange(22, 25, blueFrequency))){
+    tempColor = PURPLE;
+  }
+ 
+  if((isInRange(75, 80, redFrequency)) && (isInRange(67, 72, greenFrequency)) && (isInRange(20, 23, blueFrequency))){
+    tempColor = GREEN;
+  }
+
+    if((isInRange(58, 62, redFrequency)) && (isInRange(75, 79, greenFrequency)) && (isInRange(19, 22, blueFrequency))){
+    tempColor = ORANGE;
+  }

```

Comment:
@maikeleerens zou het mogelijk zijn om dit op te lossen met een 2-dimensionaal array? 

> Veel onleesbaarder is toch moeilijk. 😉

 
Ik denk aan:
```
Colors[
   r[min-range, max-range],
   g[min-range, max-range],
   b[min-range, max-range]
]
```
Zodat je vervolgens de actie in 1x kan testen.
Hierdoor wordt isInRange() als het goed is ook wat minder uitgevoerd, wil best [Big-O-notatie](https://en.wikipedia.org/wiki/Big_O_notation) leveren voor mijn idee als dat nodig is.

Reply:
ik zit nu 2d arrays te onderzoeken. hoe zou je dus met zo een array vergelijken? mbv for loop?